go-controller/cmd/ovnkube/ovnkube.go

```diff
@@ -579,21 +579,23 @@ func runOvnKube(ctx context.Context, runMode *ovnkubeRunMode, ovnClientset *util
 	// Note: for ovnkube node mode dpu-host no metrics is required as ovs/ovn is not running on the node.
 	if config.OvnKubeNode.Mode != types.NodeModeDPUHost && config.Metrics.OVNMetricsBindAddress != "" {
 		metricsScrapeInterval := 30
-		defer cancel()
 
 		if ovsClient == nil {
 			ovsClient, err = libovsdb.NewOVSClient(ctx.Done())
 			if err != nil {
-				return fmt.Errorf("failed to initialize libovsdb vswitchd client: %w", err)
+				ovsCLIErr = fmt.Errorf("failed to initialize libovsdb vswitchd client: %w", err)
+				cancel()
 			}
 		}
-		if config.Metrics.ExportOVSMetrics {
-			metrics.RegisterOvsMetricsWithOvnMetrics(ovsClient, metricsScrapeInterval, ctx.Done())
+		if ovsClient != nil {
```

Comment:
do we need some check for if ovsClient is nil then we also have a ovsCLI failure propagated saying client was nil? or is that an overkill?

Reply:
There is no way for ovsClient to  be nil if ovsCLIErr is not so I think we are good.

Reply:
awesome I thought the same but I asked cause you were specially checking if client  != nil so wasn't aware when and if that could even happen?